src/main/kotlin/cacophony/controlflow/FunctionHandler.kt

```diff
@@ -34,16 +42,55 @@ class GenerateVariableAccessException(
 class FunctionHandlerImpl(
     private val function: FunctionDeclaration,
     private val analyzedFunction: AnalyzedFunction,
+    // List of parents' handlers ordered from immediate parent.
+    private val ancestorFunctionHandlers: List<FunctionHandler>,
 ) : FunctionHandler {
+    private val staticLink: Variable.AuxVariable.StaticLinkVariable = Variable.AuxVariable.StaticLinkVariable()
+
     override fun getFunctionDeclaration(): FunctionDeclaration = function
 
     override fun generateCall(
         arguments: List<CFGNode>,
         result: Register?,
-    ): CFGFragment {
+    ): List<CFGNode> {
         TODO("Not yet implemented")
     }
 
+    // Wrapper for generateCall that additionally fills staticLink to parent function.
+    // Generally when we call another function we have to pass the staticLink to it,
+    // we have to find parent of called function and use variableAccess to its staticLink variable.
+    // It should be possible both from the parent itself and from nested functions,
+    // therefore there is option that both will have this staticLink,
+    // or it can be ified in such a way that calling nested functions uses RBP directly.
+    // I've decided to keep both links (to itself and to parent) in auxVariables.
+    // If it causes problems it can be modified to store only staticLink to parent.
+    override fun generateCallFrom(
+        caller: FunctionHandler,
+        arguments: List<CFGNode>,
+        result: Register?,
+    ): CFGFragment {
+        val nodes: MutableList<CFGNode> = mutableListOf()
+        if (ancestorFunctionHandlers.isNotEmpty()) {
+            val parentFunction = ancestorFunctionHandlers[0]
+            val parentLink = parentFunction.getStaticLink()
+            nodes.add(
+                CFGNode.MemoryWrite(
+                    CFGNode.MemoryAccess(caller.generateVariableAccess(parentLink)),
+                    CFGNode.MemoryAccess(this.generateVariableAccess(parentLink)),
```

Comment:
I don't understand the calling convention fully, so this is a sanity check: does it work if `caller === function`? So in case of recursion?

Reply:
If you try to generateVariableAccess to your own link now, I guess you'll should check that this variable is on the variable list of this function, so you have it and you can obtain its allocation.